### PR TITLE
Assorted doc issues

### LINKED ISSUE #10330: exclude keyword not working
I'll happily get schooled if I'm not using it correctly, but trying to "exclude" an SLS doesn't seem to work. Here is a contrived example to illustrate what I'm seeing.

**top.sls**
```
base:
  '*':
    - basic
```

**basic.sls**
```
include:
  - vim

exclude:
  - sls: vim

vim-enhanced:
  pkg.installed
```

**vim.sls**
```
vim-enhanced:
  pkg.installed
```

My understanding from the docs is that excluding vim.sls should entirely remove its contents from the highstate data before trying to evaluate it. But instead, I get the following error running a state.highstate:

```
local:
    Data failed to compile:
----------
    Detected conflicting IDs, SLS IDs need to be globally unique.
    The conflicting ID is "vim-enhanced" and is found in SLS "base:basic" and SLS "base:vim"
```

```
salt-call --versions-report
           Salt: 0.17.5
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: unknown
       M2Crypto: 0.20.2
 msgpack-python: 0.1.13
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
         PyYAML: 3.10
          PyZMQ: 2.2.0.1
            ZMQ: 3.2.4
```

### LINKED ISSUE #29520: [doc] transport option not in default master configuration nor in the example file of the documentation
https://docs.saltstack.com/en/latest/topics/releases/2015.8.0.html talks about zmq/raet/tcp (and tornado). https://docs.saltstack.com/en/latest/topics/transports/raet/index.html talks about "transport" in the master config. 

https://docs.saltstack.com/en/latest/ref/configuration/examples.html#configuration-examples-master does not have this configuration attribute.

### LINKED ISSUE #29528: nitpick for "Using Salt at scale" tutorial
https://docs.saltstack.com/en/latest/topics/tutorials/intro_scale.html#too-many-minions-returning-at-once says you can avoid all the minions returning at once to a '*' test.ping by batching the command.

This is wrong, because batching is done by gathering the list of minions to use in a batch, which is done with a test.ping to the target (here, it would be '*'), without batching obviously.  See Batch.__gather_minions(). So that example needs to be changed.

It would be nice to have batching in some of those wildcard test.pings that are found here and there in the code for the salt master but that's another issue.

### LINKED ISSUE #29636: Documentation for the refresh_password arg in postgres_user.present is confusing
The documentation says:

> If refresh_password is None or False, the password will be automatically updated without extra password change check.

> This behaviour makes it possible to execute in environments without superuser access available, e.g. Amazon RDS for PostgreSQL

In my experience, and looking at the code, the behavior is actually the reverse.

If you want the password to be updated without a check (e.g. on RDS), you need to set `refresh_password` to True.

The same is probably true for `postgres_group.present`.

See:

- https://docs.saltstack.com/en/latest/ref/states/all/salt.states.postgres_user.html#salt.states.postgres_user.present
- https://docs.saltstack.com/en/latest/ref/states/all/salt.states.postgres_group.html#salt.states.postgres_group.present

### LINKED ISSUE #30261: Add ability to define custom beacons
It would be nice if you where able to create your own custom beacons, just as you can with modules and runners.

I've searched the docs, and can't see any way to do this yet.

### LINKED ISSUE #30464: pillar_env minion config option needs to be documented
As per #28237, there is a new 'pillarenv' config option - but it's not in any of the sample Minion configuration documentation.

Need to make sure people are aware that 'environment' doesn't set the pillarenv anymore and only works for the fileserver.

### LINKED ISSUE #31216: pkg.installed documentation not consistent with implementation
The documentation for pkg.installed says:

> version (str) --
> Install a specific version of a package. This option is ignored if either "pkgs" or "sources" is used.

I'm running 2015.8.5 on CentOS and the version parameter is not ignored if I use 'pkgs' like this in my formula:
```
common_packages:
  pkg.installed:
    - pkgs:
      - unzip
      - dos2unix
      - salt-minion: 2015.8.5-1.el6
```

Here's the log:
```
 ID: common_packages
    Function: pkg.installed
      Result: True
     Comment: 3 targeted packages were installed/updated. The following packages were already installed: salt-minion=2015.8.5-1.el6
     Started: 11:52:51.306751
    Duration: 27051.788 ms
     Changes:
              ----------
              dos2unix:
                  ----------
                  new:
                      3.1-37.el6
                  old:
              unzip:
                  ----------
                  new:
                      6.0-2.el6_6
                  old:
```
I suspect the documentation needs to be updated.

### PR DESCRIPTION
**Pull request by jacobhammons**
Fixes #10330
Fixes #31216
Fixes #30464
Fixes #29520
Fixes #30261
Fixes #29636
Fixes #29528

### CODE REVIEW

File: doc/ref/configuration/master.rst [hunk 1]
```diff
@@ -494,6 +494,23 @@ that connect to a master via localhost.
...
+.. conf_master:: transport
+
+``transport``
+-------------
+
+Default: ``zeromq``
+
+Changes the underlying transport layer. Zeromq is the recommended transport
+while additional transport layers are under development. Supported values are
+``zeromq``, ``raet`` (experimental), and ``tcp`` (experimental). This settings has
```
Comment (cachedout) [inline doc/ref/configuration/master.rst]: 'This settings`?

File: doc/ref/configuration/master.rst [hunk 2]
```diff
@@ -494,6 +494,23 @@ that connect to a master via localhost.
...
     presence_events: False
 
+.. conf_master:: transport
+
+``transport``
+-------------
+
+Default: ``zeromq``
+
+Changes the underlying transport layer. Zeromq is the recommended transport
```
Comment (cachedout) [inline doc/ref/configuration/master.rst]: `ZeroMQ` not 'Zeromq'.

File: doc/ref/configuration/minion.rst [hunk 3]
```diff
@@ -578,6 +578,23 @@ Pull port used when :conf_minion:`ipc_mode` is set to ``tcp``.
...
     tcp_pull_port: 4511
 
+.. conf_minion:: transport
+
+``transport``
+-------------
+
+Default: ``zeromq``
+
+Changes the underlying transport layer. Zeromq is the recommended transport
```
Comment (cachedout) [inline doc/ref/configuration/minion.rst]: Same here for `Zeromq and 'This settings`

**Comment by jacobhammons** [comment]
@cachedout see https://github.com/saltstack/salt/pull/31485, let me know if I fixed that right. Thanks!

**Pull request merged**